Filtering: Include everything by default

martinRenou opened the pull request:
Towards fixing https://github.com/jupyterlite/xeus/issues/66

This PR updates the empack config to include everything by default, and only extra exclude rules should be added for packages.

I believe the original filtering implementation is good, it's our usage of the config that was not user-friendly nor practical.

martinRenou commented:
@DerThorsten Would you like to have a look? It's unclear to me why the CI fails, it looks like there is a timeout error on the playwright tests

martinRenou commented:
It seems xeus-python cannot start when using this empack branch, this may be why the playwright test fails

![Screenshot from 2024-06-05 17-54-47](https://github.com/emscripten-forge/empack/assets/21197331/ca6ef2cf-df95-4d7d-b193-a4ed0a5eddc0)

DerThorsten commented:
> It seems xeus-python cannot start when using this empack branch, this may be why the playwright test fails
> 
> ![Screenshot from 2024-06-05 17-54-47](https://private-user-images.githubusercontent.com/21197331/336915209-ca6ef2cf-df95-4d7d-b193-a4ed0a5eddc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2MDMzMjgsIm5iZiI6MTcxNzYwMzAyOCwicGF0aCI6Ii8yMTE5NzMzMS8zMzY5MTUyMDktY2E2ZWYyY2YtZGY5NS00ZDdkLWIxOTMtYTRlZDBhNWVkZGMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDE1NTcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhNmU5OWRjNmY2OTE1ZjkwM2ZlN2Q3NzM4ZTdiOTQxNThmZmJjNmYzMTEzY2EzZmNmZTk2MGI2ZWU5OGIwZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CqQAWD1po0MwDwOJOs8tiSizKD4Hjn0SIXTHUzGNj6c)

I will debug this tomorrow, thanks for starting this!

martinRenou commented:
I'm reworking the implementation to completely ditch the `include_patterns` parameter

martinRenou commented:
Yep, that was what I implemented initially but I've done it in the default config. I just reworked the implementation to do this in the Python code. That way if users are already shipping their own config they will get the fix too.

martinRenou commented:
> I will debug this tomorrow, thanks for starting this!

I should be able to debug this, I'm looking at the diff between the packed env on `main` and the packed env on this branch. It seems to differ a lot because I'm now including lots of stuff which may prevent it to start.

martinRenou commented:
The CI failure should be fixed by https://github.com/emscripten-forge/pyjs/pull/67
